src/wallet/rpcnames.cpp

```diff
@@ -523,6 +633,10 @@ name_firstupdate ()
       throw JSONRPCError (RPC_TRANSACTION_ERROR, "previous txid not found");
   }
 
+  // If TXID is fixed, we won't do the search above.
+  if (fixedTxid && !fixedRand)
+    getNameSalt (pwallet, name, prevOut.scriptPubKey, rand);
```

Comment:
What about just always determining the `rand` value here `if(!fixedRand)` instead of in the txid lookup code?  I think this would make the code simpler to read and understand.  Also, you need to check the return value of `getNameSalt` and throw some RPC error if the lookup fails. Perhaps even then check again that the `rand` we recovered actually matches the hash in the previous output as well, so we can fail here with a proper error instead of the tx just being invalid later.  (This rand-to-hash check could be factored out into a helper function, so it can be reused in the txid lookup and here.)

Also, I think it would make sense to add an explicit test to the regtest what happens if you pass an explicit but wrong txid (i.e. for another name).

Reply:
But then I have to determine rand twice, first inside the loop and throw it away, and then outside. That's a little bit unseemly.

Might do an helper function, yes. Could be used in `name_new` too, right?

Yes, will add. Maybe something for nonexistent TXID too?

Reply:
True, but I think determining it twice is not a big deal.  There is not really a performance concern here IMHO, and it will make the code cleaner to read, which is more important than a non-bottleneck performance increase.

Reply:
Does it, though? It makes the code more complex.

Eh, whatever. If you say so.